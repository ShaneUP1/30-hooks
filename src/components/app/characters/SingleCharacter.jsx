import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const SingleCharacter = ({ name, species, image, id }) => (
  <figure>
    <Link to={`details/${id}`} >
      <img src={image} alt={name} />
    </Link>
    <figcaption>
      <h2>{name}</h2>
      <p>{species}</p>
    </figcaption>
  </figure>
);

SingleCharacter.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default SingleCharacter;
