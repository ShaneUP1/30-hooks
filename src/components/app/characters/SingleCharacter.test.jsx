import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SingleCharacter from './SingleCharacter';

describe('SingleCharacter component', () => {
  afterEach(() => cleanup());
  it('renders a single character', () => {
    const { asFragment } = render(<SingleCharacter
      name={'shane'} image={'www.url.com'} species={'toad'}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
