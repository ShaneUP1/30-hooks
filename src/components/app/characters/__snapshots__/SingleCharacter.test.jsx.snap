// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleCharacter component renders a single character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="shane"
      src="www.url.com"
    />
    <figcaption>
      <h2>
        shane
      </h2>
      <p>
        toad
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
